src/main/java/org/mapfish/print/servlet/MapPrinterServlet.java

```diff
@@ -147,7 +147,7 @@ protected void createAndGetPDF(HttpServletRequest httpServletRequest, HttpServle
         } catch (UnsupportedEncodingException e) {
             throw new RuntimeException(e);
         }
-        if (httpServletRequest.getMethod() == "POST") {
+        if (httpServletRequest.getMethod().equalsIgnoreCase("POST")) {
```

Comment:
Fair enough, do we have any assurance that getMethod() returns non null? or do we need a null check here?

It is usually safer in java code to do reviser order to protect against NPE:
```
if ("POST".equalsIgnoreCase(httpServletRequest.getMethod()) {
```